producerConsumer.java

```diff
@@ -0,0 +1,105 @@
+package publisherSubscriber;
+
+import java.util.*;
+
+class sharedResource {
+
+    int size = 5;
+    Queue<Integer> queue = new ArrayDeque<>(size);
+
+    public boolean isFull() {
+        return queue.size() >= 5;
+    }
+
+    public boolean isEmpty() {
```

Comment:
Consider making `isEmpty` a private method, as this is not used outside of this class